external-reports/sensors/src/main/java/org/sonarsource/plugins/externalreport/checkstyle/CheckstyleSensor.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.plugins.externalreport.checkstyle;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import javax.xml.stream.XMLStreamException;
+import org.sonar.api.Plugin.Context;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.batch.sensor.issue.NewExternalIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.analyzer.commons.ExternalReportProvider;
+import org.sonarsource.analyzer.commons.ExternalRuleLoader;
+import org.sonarsource.plugins.externalreport.ExternalReportExtensions;
+import org.sonarsource.plugins.externalreport.commons.ExternalRulesDefinition;
+
+public class CheckstyleSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(CheckstyleSensor.class);
+
+  private static final String LINTER_KEY = "checkstyle";
+
+  private static final String LINTER_NAME = "Checkstyle";
+
+  private static final String LANGUAGE_KEY = "java";
+
+  private static final String REPORT_PROPERTY_KEY = "sonar.java.checkstyle.reportPaths";
+
+  private static class SingletonRuleLoader {
+    private static final ExternalRuleLoader INSTANCE = new ExternalRuleLoader(
+      CheckstyleSensor.LINTER_KEY,
+      CheckstyleSensor.LINTER_NAME,
+      "org/sonar/l10n/java/rules/checkstyle/rules.json",
+      CheckstyleSensor.LANGUAGE_KEY);
+  }
+
+  static ExternalRuleLoader ruleLoader() {
+    return SingletonRuleLoader.INSTANCE;
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor
+      .onlyOnLanguage(CheckstyleSensor.LANGUAGE_KEY)
+      .onlyWhenConfiguration(conf -> conf.hasKey(REPORT_PROPERTY_KEY))
+      .name("Import of Checkstyle issues");
+  }
+
+  public static void define(Context context, boolean externalIssuesSupported) {
```

Comment:
In fact, why not removing the boolean parameter and move the `if` to the caller (`ExternalReportExtensions`)?

Reply:
because "context.addExtension(CheckstyleSensor.class);" is always added. To be able to log:
LOG.error("Import of external issues requires SonarQube 7.2 or greater.");
see method ExternalReportProvider.getReportFiles(...)

Reply:
Ok, I can do SonarTech proposal and create two methods